RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc

```diff
@@ -424,51 +303,63 @@ double DAClusterizerInZ_vect::updateTc(double beta, track_t& gtracks, vertex_t&
                                           vertex_t const& vertices,
                                           const unsigned int kmin,
                                           const unsigned int kmax) {
-    const double track_z = tracks.z_ptr[itrack];
-    const double botrack_dz2 = -beta * tracks.dz2_ptr[itrack];
+    const double track_z = tracks.zpca[itrack];
+    const double botrack_dz2 = -beta * tracks.dz2[itrack];
 
     // auto-vectorized
     for (unsigned int ivertex = kmin; ivertex < kmax; ++ivertex) {
-      auto mult_res = track_z - vertices.z_ptr[ivertex];
-      vertices.ei_cache_ptr[ivertex] = botrack_dz2 * (mult_res * mult_res);
+      auto mult_res = track_z - vertices.zvtx[ivertex];
+      vertices.exp_arg[ivertex] = botrack_dz2 * (mult_res * mult_res);
     }
   };
 
   auto kernel_add_Z_range = [Z_init](
                                 vertex_t const& vertices, const unsigned int kmin, const unsigned int kmax) -> double {
     double ZTemp = Z_init;
     for (unsigned int ivertex = kmin; ivertex < kmax; ++ivertex) {
-      ZTemp += vertices.pk_ptr[ivertex] * vertices.ei_ptr[ivertex];
+      ZTemp += vertices.rho[ivertex] * vertices.exp[ivertex];
     }
     return ZTemp;
   };
 
-  auto kernel_calc_normalization_range = [beta](const unsigned int track_num,
-                                                track_t& tks_vec,
-                                                vertex_t& y_vec,
-                                                const unsigned int kmin,
-                                                const unsigned int kmax) {
-    auto tmp_trk_pi = tks_vec.pi_ptr[track_num];
-    auto o_trk_Z_sum = 1. / tks_vec.Z_sum_ptr[track_num];
-    auto o_trk_dz2 = tks_vec.dz2_ptr[track_num];
-    auto tmp_trk_z = tks_vec.z_ptr[track_num];
+  auto kernel_calc_normalization_range = [beta, updateTc](const unsigned int track_num,
+                                                          track_t& tracks,
+                                                          vertex_t& vertices,
+                                                          const unsigned int kmin,
+                                                          const unsigned int kmax) {
+    auto tmp_trk_tkwt = tracks.tkwt[track_num];
+    auto o_trk_sum_Z = 1. / tracks.sum_Z[track_num];
+    auto o_trk_dz2 = tracks.dz2[track_num];
+    auto tmp_trk_z = tracks.zpca[track_num];
     auto obeta = -1. / beta;
```

Comment:
This line can be moved inside the following loop

Reply:
I guess you mean inside the if(useTc) , outside the loop. 
What's that about the evalF? I thought I had removed it and don't see t anymore.

Reply:
> I guess you mean inside the if(useTc) , outside the loop.

Yes (loop -> block,: sorry for the typo), to avoid computing it when not needed

> What's that about the evalF? I thought I had removed it and don't see t anymore.

Right, my bad: it is not there any more, I probably looked at an outdated version of the code